test/serializers/associations_test.rb

```diff
@@ -125,6 +125,26 @@ def test_associations_custom_keys
         assert expected_association_keys.include? :site
       end
 
+      class BelongsToTestPostSerializer < ActiveModel::Serializer
+        belongs_to :blog
+      end
+
+      def test_belongs_to_doesnt_load_not_included_record_for_jsonapi
+        post = Post.new(blog_id: 'the_blog_id')
+        def post.blog
+          fail 'should use blog_id'
+        end
+        def post.read_attribute_for_serialization(name)
+          # Bug in how ActiveModelSerializers::Model handles attributes and methods
+          send(name)
+        end
```

Comment:
The current impl is

```ruby
    def read_attribute_for_serialization(key)
      if key == :id || key == 'id'
        attributes.fetch(key) { id }
      else
        attributes[key]
      end
    end
```

Which doesn't send and relies overly much on the attributes method, which is only set at initialize... This is definitely a bug that needs to be fixed.....

Until I wrote this, the test was't raising the error